src/App.js

```diff
@@ -1,14 +1,30 @@
-import React from 'react';
+import React, {useCallback, useState} from 'react';
+import {Switch, BrowserRouter, Route, Redirect} from 'react-router-dom'
 import Header from 'pages/shared/Header/Header';
 import Footer from 'pages/shared/Footer/Footer';
 import Home from 'pages/Home/Home';
+import MovieDetails from "pages/MovieDetails/MovieDetails";
 
-const App = () => (
-    <>
-        <Header/>
-        <Home/>
-        <Footer/>
-    </>
-);
+const App = () => {
+
+    const [searchIconVisible, setShowSearchIconVisible] = useState(true);
+
+    const handleSearchIcon = useCallback(() => {
+        console.log("change button");
+        setShowSearchIconVisible(!searchIconVisible);
+    }, [searchIconVisible]);
+
```

Comment:
I think in this case it would be great to use SearchIcon and MovieItem callback onClick.

For example:
 `const [searchIconVisible, setShowSearchIconVisible] = useState(false);

    const handleSearchIcon = (visibility) => {
        setShowSearchIconVisible(visibility);
    }`
    
    and in searchIcon set new state:
    `const goToSearch = () => {
        history.push(`/`)
        handleSearchIcon(false);
    }`

    It allows you to control current state

Reply:
Hi Natalia,
Yep, you are right, and it's much better and clear approach. Updated.